### PR TITLE
#2828 Fixed welders not turning off after spilling contents.

### PR DESCRIPTION
**Pull request by Necromunger**
# Pull Request Template

### Purpose

#2828 Fixed welders not turning off after spilling contents.

### Please make sure you have followed the self checks below before submitting a PR:

- [x] Code is sufficiently commented
- [x] Code is indented with tabs and not spaces
- [x]  The PR does not include any unnecessary .meta, .prefab or <b>.unity (scene) changes</b>
- [x]  The PR does not bring up any new compile errors
- [x]  The PR has been tested in editor
- [x]  Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [x]  Any new / changed components follow the [Component Development Checklist]
- [x]  Any new objects / items follow the [Creating Items and Objects Guide]
- [x]  The PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [x]  The PR has been tested with round restarts.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Chemistry/ReagentContainer.cs [hunk 1]
```diff
@@ -376,19 +379,25 @@ private void SpillAll(Vector3Int worldPos, bool thrown = false)
...
 		{
-			Chat.AddLocalMsgToChat($"{gameObject.ExpensiveName()}'s contents spill all over the floor!",(Vector3)worldPos, gameObject);
+			Chat.AddLocalMsgToChat($"{gameObject.ExpensiveName()}'s contents spill all over the floor!", (Vector3)worldPos, gameObject);
 		}
 
 		var spilledReagents = TakeReagents(AmountOfReagents(Contents));
 		MatrixManager.ReagentReact(spilledReagents, worldPos);
 
-	//todo: half decent regs spread
+		var welder = gameObject.GetComponent<Welder>();
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Chemistry/ReagentContainer.cs]: this seems backwards. I’d rather introduce the reagent level UnityEvent in ReagentContainer and subscribe to it from within the welder script

Comment (Necromunger) [reply UnityProject/Assets/Scripts/Chemistry/ReagentContainer.cs]: Will make your suggested change, i agree that would be a better idea.

**Comment by Necromunger** [comment]
@fomalsd UnityEvents implemented to have other scripts listen to the spill event.

**Comment by daalta** [comment]
Hey, yo, thanks for fixing this!

**Review by fomalsd — APPROVED** [review]

**Pull request merged**